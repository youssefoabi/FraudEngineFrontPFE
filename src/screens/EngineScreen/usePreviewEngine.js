import { useCallback } from 'react';
import { useDispatch } from 'react-redux';

import { PREVIEW_ENGINE } from '../../components/EngineForm/actions';

export default function usePreviewEngine() {
  const dispatch = useDispatch();

  const previewEngine = useCallback(id => dispatch({ type: PREVIEW_ENGINE, payload: { id } }), [dispatch]);

  return { previewEngine };
}
