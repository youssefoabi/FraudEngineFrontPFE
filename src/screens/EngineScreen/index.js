import { Button, Footer, Screen } from '@cdiscount/backoffice-ui';
import { func, shape, string } from 'prop-types';
import { pathOr } from 'ramda';
import { Save, Visibility } from '@material-ui/icons';
import React from 'react';

import EngineForm from '../../components/EngineForm';
import ServerActionBackdrop from '../../components/ServerActionBackdrop';
import TitleBar from '../../components/TitleBar';
import usePreviewEngine from './usePreviewEngine';

const EngineScreen = ({ engine, setEngine, action, title }) => {
  const { previewEngine } = usePreviewEngine();
  const name = pathOr('', ['name'], engine);
  const id = pathOr(null, ['id'], engine);

  return (
    <Screen
      titlebar={
        <TitleBar
          title={`${title} d'un Moteur de Recherche : ${name || ''}`}
          actions={[
            {
              id: 'preview_engine_button',
              label: 'PrÃ©visualiser',
              icon: Visibility,
              size: 'large',
              onClick: () => previewEngine(id),
            },
            {
              id: 'save_engine_button',
              label: 'Enregistrer',
              icon: Save,
              size: 'large',
              onClick: () => action(engine),
            },
          ]}
        />
      }
      bottombar={
        <Footer>
          <Button
            onClick={() => action(engine)}
            startIcon={<Save />}
            size="large"
            color="tertiary"
            label="Enregistrer"
            variant="contained"
          />
        </Footer>
      }
    >
      <EngineForm engine={engine} setEngine={setEngine} />
      <ServerActionBackdrop />
    </Screen>
  );
};

EngineScreen.propTypes = {
  engine: shape({}),
  title: string,
  action: func,
  setEngine: func,
};

EngineScreen.defaultProps = {
  engine: null,
  title: '',
  action: Function.Prototype,
  setEngine: Function.Prototype,
};

export default EngineScreen;
