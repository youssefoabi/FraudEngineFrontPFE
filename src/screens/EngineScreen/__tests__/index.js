import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import EngineScreen from '../';

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
}));

describe(EngineScreen.name, () => {
  itRendersCorrectly(<EngineScreen />);
});
