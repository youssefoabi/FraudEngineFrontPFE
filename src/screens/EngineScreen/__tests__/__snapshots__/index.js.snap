// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngineScreen renders correctly 1`] = `
<o
  bottombar={
    <p
      backToTopLabel="Haut de page"
    >
      <y
        color="tertiary"
        icon={null}
        label="Enregistrer"
        onClick={[Function]}
        size="large"
        startIcon={<UNDEFINED />}
        variant="contained"
      />
    </p>
  }
  titlebar={
    <withRouter(TitleBar)
      actions={
        Array [
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "VisibilityIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": "preview_engine_button",
            "label": "PrÃ©visualiser",
            "onClick": [Function],
            "size": "large",
          },
          Object {
            "icon": Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "SaveIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "id": "save_engine_button",
            "label": "Enregistrer",
            "onClick": [Function],
            "size": "large",
          },
        ]
      }
      title=" d'un Moteur de Recherche : "
    />
  }
>
  <EngineForm
    engine={null}
    setEngine={[Function]}
  />
  <ServerActionBackdrop />
</o>
`;
